presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomFilters.java

```diff
@@ -0,0 +1,333 @@
+        }
+    }
+
+    @Test
+    // simulate query on a 2 columns where 1 is used as part of the where, with and without bloom filter
+    public void testMatches()
+            throws Exception
+    {
+        // stripe columns
+        TestingColumnHandle testingColumnHandle = new TestingColumnHandle("bigint_0");
+        Domain testingColumnHandleDomain = Domain.singleValue(BIGINT, 1234L);
+        TupleDomain.ColumnDomain<TestingColumnHandle> column = new TupleDomain.ColumnDomain<>(testingColumnHandle, testingColumnHandleDomain);
+
+        TestingColumnHandle testingColumnHandle1 = new TestingColumnHandle("bigint_1");
+        Domain testingColumnHandleDomain1 = Domain.singleValue(BIGINT, 234L);
+        TupleDomain.ColumnDomain<TestingColumnHandle> column1 = new TupleDomain.ColumnDomain<>(testingColumnHandle1, testingColumnHandleDomain1);
+
+        // predicate consist of the bigint_0 = 1234
+        TupleDomain<TestingColumnHandle> effectivePredicate = TupleDomain.fromColumnDomains(Optional.<List<TupleDomain.ColumnDomain<TestingColumnHandle>>>of(ImmutableList.of(column)));
+        TupleDomain<TestingColumnHandle> emptyEffectivePredicate = TupleDomain.fromColumnDomains(Optional.<List<TupleDomain.ColumnDomain<TestingColumnHandle>>>of(ImmutableList.of()));
+
+        // predicate column references
+        ImmutableList.Builder<TupleDomainOrcPredicate.ColumnReference<TestingColumnHandle>> columnReferences = ImmutableList.builder();
+        columnReferences.add(new TupleDomainOrcPredicate.ColumnReference<>(testingColumnHandle, 0, BIGINT));
+        columnReferences.add(new TupleDomainOrcPredicate.ColumnReference<>(testingColumnHandle1, 1, BIGINT));
+
+        TupleDomainOrcPredicate predicate = new TupleDomainOrcPredicate<>(effectivePredicate, columnReferences.build(), true);
+        TupleDomainOrcPredicate emptyPredicate = new TupleDomainOrcPredicate<>(emptyEffectivePredicate, columnReferences.build(), true);
+
+        // assemble a matching and a non-matching bloom filter
+        HiveBloomFilter hiveBloomFilter = new HiveBloomFilter(new BloomFilter(1000, 0.01D));
+        OrcProto.BloomFilter emptyOrcBloomFilter = toOrcBloomFilter(hiveBloomFilter);
+        hiveBloomFilter.addLong(1234L);
+        OrcProto.BloomFilter orcBloomFilter = toOrcBloomFilter(hiveBloomFilter);
+
+        ImmutableMap.Builder<Integer, ColumnStatistics> matchingStatisticsBuilder = ImmutableMap.<Integer, ColumnStatistics>builder();
+        matchingStatisticsBuilder.put(0, new ColumnStatistics(null, null, new IntegerStatistics(10L, 2000L), null, null, null, null, new HiveBloomFilter(orcBloomFilter.getBitsetList(), orcBloomFilter.getBitsetCount() * 64, orcBloomFilter.getNumHashFunctions())));
+        Map<Integer, ColumnStatistics> matchingStatisticsByColumnIndex = matchingStatisticsBuilder.build();
+
+        ImmutableMap.Builder<Integer, ColumnStatistics> nonMatchingStatisticsBuilder = ImmutableMap.<Integer, ColumnStatistics>builder();
```

Comment:
Since you are using only element in this maps, just the simple form:
```java
        Map<Integer, ColumnStatistics> matchingStatisticsByColumnIndex = ImmutableMap.of(0, new ColumnStatistics(
                null,
                null,
                new IntegerStatistics(10L, 2000L),
                null,
                null,
                null,
                null,
                new HiveBloomFilter(orcBloomFilter.getBitsetList(), orcBloomFilter.getBitsetCount() * 64, orcBloomFilter.getNumHashFunctions())));